src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java

```diff
@@ -1215,18 +1214,26 @@ private DraftOrder getCurrentUploadDraftOrderDetails(CaseData caseData, String l
                               .build())
             .dateOrderMade(caseData.getDateOrderMade())
             .approvalDate(caseData.getApprovalDate())
-            .judgeNotes(caseData.getManageOrders() != null
-                            ? caseData.getManageOrders().getJudgeDirectionsToAdminAmendOrder() : null)
+            .judgeNotes(caseData.getJudgeDirectionsToAdmin())
             .orderSelectionType(orderSelectionType)
             .orderCreatedBy(loggedInUserType)
-            .isOrderUploadedByJudgeOrAdmin(null != caseData.getManageOrdersOptions()
-                                               && caseData.getManageOrdersOptions().equals(uploadAnOrder) ? Yes : No)
+            .isOrderUploadedByJudgeOrAdmin(getIsUploadedFlag(caseData.getManageOrdersOptions(), loggedInUserType))
             .manageOrderHearingDetails(caseData.getManageOrders().getOrdersHearingDetails())
             .hasJudgeProvidedHearingDetails(caseData.getManageOrders().getHasJudgeProvidedHearingDetails())
             .isOrderCreatedBySolicitor(UserRoles.SOLICITOR.name().equals(loggedInUserType) ? Yes : No)
             .build();
     }
 
+    private YesOrNo getIsUploadedFlag(ManageOrdersOptionsEnum manageOrdersOptions, String loggedInUserType) {
+        if (UserRoles.SOLICITOR.name().equals(loggedInUserType)) {
+            return Yes;
+        } else if (null != manageOrdersOptions
+            && manageOrdersOptions.equals(uploadAnOrder)) {
```

Comment:
assuming uploadAnOrder is a constant? if yes can we alter the if logic?

Reply:
And this if else logic will throw Major code smell. We should have a variable holding the return value and returning it.